### PR TITLE
Store grids in a DynamicTree

### PR DESCRIPTION
**Pull request by metalgearsloth**
Removes the nasty O(n) check from FindGridsIntersecting which is much better for large grid counts.

I wasted a lot of time trying to debug issues until I realised ClientMapManager never applied the grid's rotation.

TODO:

- [x] Anchoring test for rotated grid (I think this already gets caughhhttt?)
- [x] Move grids off of the map's broadphase

### CODE REVIEW

**Comment by moonheart08** [comment]
merg when

File: Robust.Client/GameController/GameController.cs [hunk 1]
```diff
@@ -134,8 +134,8 @@ internal bool StartupContinue(DisplayMode displayMode)
             _prototypeManager.Initialize();
             _prototypeManager.LoadDirectory(Options.PrototypeDirectory);
             _prototypeManager.Resync();
-            _mapManager.Initialize();
             _entityManager.Initialize();
+            _mapManager.Initialize();
```
Comment (mirrorcult) [inline Robust.Client/GameController/GameController.cs:138]: this seems sus, whats the rationale

**Review by mirrorcult — APPROVED** [review]

File: Robust.Client/GameController/GameController.cs (hunk 1 above)
Comment (metalgearsloth) [reply Robust.Client/GameController/GameController.cs:138]: Eventbus isn't instantiated on entitymanager until then so mapmanager can't subscribe to events (this is also the order the server initialises them in).

Comment (mirrorcult) [reply Robust.Client/GameController/GameController.cs:138]: ok sgtm

**Pull request merged**